airflow/providers/google/cloud/operators/compute.py

```diff
@@ -240,79 +729,358 @@ def __init__(
             **kwargs,
         )
 
-    def _validate_all_body_fields(self) -> None:
-        if self._field_validator:
-            self._field_validator.validate(self.body)
-
+    def _validate_all_body_fields(self) -> None:
+        if self._field_validator:
+            self._field_validator.validate(self.body)
+
+    def execute(self, context: 'Context') -> None:
+        hook = ComputeEngineHook(
+            gcp_conn_id=self.gcp_conn_id,
+            api_version=self.api_version,
+            impersonation_chain=self.impersonation_chain,
+        )
+        self._validate_all_body_fields()
+        ComputeInstanceDetailsLink.persist(
+            context=context,
+            task_instance=self,
+            location_id=self.zone,
+            resource_id=self.resource_id,
+            project_id=self.project_id or hook.project_id,
+        )
+        return hook.set_machine_type(
+            zone=self.zone, resource_id=self.resource_id, body=self.body, project_id=self.project_id
+        )
```

Comment:
Should we use `return` if we do not return any? In my opinion it is better to call this function without return

Reply:
Done